backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java

```diff
@@ -116,16 +116,39 @@ public DatasetVersionServiceImpl(
           new ArrayList<>(datasetIdSet).get(0));
       accessibleDatasetVersionIds.addAll(requestedDatasetVersionIds);
     } else {
-      allowedDatasetIds =
+      allowedDatasetVersionIds =
           roleService.getSelfAllowedResources(
               ModelDBServiceResourceTypes.DATASET, modelDBServiceActions);
       // Validate if current user has access to the entity or not
-      allowedDatasetIds.retainAll(requestedDatasetVersionIds);
-      accessibleDatasetVersionIds.addAll(allowedDatasetIds);
+      allowedDatasetVersionIds.retainAll(datasetIdSet);
```

Comment:
datasetIdSet ? is this right? if so was there a bug to begin with?

Reply:
it was a typo. fixed

Reply:
I do not see this changed